pkg/skaffold/schema/latest/v1/config.go

```diff
@@ -1208,6 +1208,11 @@ type DockerArtifact struct {
 	// For example: `{"key1": "value1", "key2": "value2"}`.
```

Comment:
This example should include a templated field.

Reply:
The latest commit does this.  I've also removed the schema change to DockerArtifact.

Reply:
There's an issue with modifying [pkg/skaffold/schema/latest/v1/config.go](./pkg/skaffold/schema/latest/v1/config.go), since it annoys one of the "check" scripts. Doesn't want it modified since latest is assigned to an already released version.

Reply:
"but Docker does not propagate environment variables into the container: that's the purpose of the build-args."  The use case here is NextJS, which has "magic" environment variables that get built into server-side rendereding containers. Which is a PITA. But it is what it is.

Reply:
Build now passes; @briandealwis's requested changes have been made.